components/misc_metrics/language_metrics.cc

```diff
@@ -0,0 +1,263 @@
+        "ru",  // Russian
+        "rw",  // Kinyarwanda
+        "sa",  // Sanskrit
+        "sc",  // Sardinian
+        "sd",  // Sindhi
+        "se",  // Northern Sami
+        "sg",  // Sango
+        "si",  // Sinhala; Sinhalese
+        "sk",  // Slovak
+        "sl",  // Slovenian
+        "sm",  // Samoan
+        "sn",  // Shona
+        "so",  // Somali
+        "sq",  // Albanian
+        "sr",  // Serbian
+        "ss",  // Swati
+        "st",  // "Sotho, Southern"
+        "su",  // Sundanese
+        "sv",  // Swedish
+        "sw",  // Swahili
+        "ta",  // Tamil
+        "te",  // Telugu
+        "tg",  // Tajik
+        "th",  // Thai
+        "ti",  // Tigrinya
+        "tk",  // Turkmen
+        "tl",  // Tagalog
+        "tn",  // Tswana
+        "to",  // Tonga (Tonga Islands)
+        "tr",  // Turkish
+        "ts",  // Tsonga
+        "tt",  // Tatar
+        "tw",  // Twi
+        "ty",  // Tahitian
+        "ug",  // Uighur; Uyghur
+        "uk",  // Ukrainian
+        "ur",  // Urdu
+        "uz",  // Uzbek
+        "ve",  // Venda
+        "vi",  // Vietnamese
+        "vo",  // Volapük
+        "wa",  // Walloon
+        "wo",  // Wolof
+        "xh",  // Xhosa
+        "yi",  // Yiddish
+        "yo",  // Yoruba
+        "za",  // Zhuang; Chuang
+        "zh",  // Chinese
+        "zu",  // Zulu
+    });
+
+constexpr LanguageSynonymPair kLanguageSynonyms[] = {
+    {"in", "id"}, {"iw", "he"}, {"ji", "yi"}, {"jw", "jv"}, {"mo", "ro"},
+};
+
+LanguageMetrics::LanguageMetrics(PrefService* profile_prefs)
+    : profile_prefs_(profile_prefs) {
+  pref_change_registrar_.Init(profile_prefs);
+  pref_change_registrar_.Add(
+      language::prefs::kAcceptLanguages,
+      base::BindRepeating(&LanguageMetrics::RecordLanguageMetric,
+                          base::Unretained(this)));
+
+  RecordLanguageMetric();
+}
+
+LanguageMetrics::~LanguageMetrics() = default;
+
+void LanguageMetrics::RecordLanguageMetric() {
+  auto languages = profile_prefs_->GetString(language::prefs::kAcceptLanguages);
+  auto languages_split = base::SplitString(
+      languages, ",", base::WhitespaceHandling::TRIM_WHITESPACE,
+      base::SplitResult::SPLIT_WANT_NONEMPTY);
+  if (languages_split.empty()) {
+    // Suspend metric if primary language not detected
+    UMA_HISTOGRAM_EXACT_LINEAR(kPrimaryLanguageHistogramName, INT_MAX - 1,
+                               kOrderedLangCodes.size());
+    return;
+  }
+
+  auto primary_language = language::ExtractBaseLanguage(languages_split[0]);
+  for (const auto& synonym_pair : kLanguageSynonyms) {
+    if (primary_language == synonym_pair.deprecated_code) {
+      primary_language = synonym_pair.active_code;
+    }
+  }
+
+  auto* language_it = kOrderedLangCodes.find(primary_language);
+  if (language_it == kOrderedLangCodes.end()) {
+    // language is not in list, suspend metric
+    UMA_HISTOGRAM_EXACT_LINEAR(kPrimaryLanguageHistogramName, INT_MAX - 1,
```

Comment:
(https://chromium.googlesource.com/chromium/src.git/+/HEAD/tools/metrics/histograms/README.md#don_t-use-the-same-histogram-logging-call-in-multiple-places)